test/spec/authn/mfa-challenge.js

```diff
@@ -1522,6 +1548,162 @@ describe('MFA_CHALLENGE', function () {
+        });
+        
+        context.transaction = oktaAuth.tx.createTransaction(mfaPush.response);
+      });
+
+      afterEach(() => {
+        jest.runOnlyPendingTimers();
+        jest.useRealTimers();
+      });
+
+      it('should proceed with flow as normal if document is never hidden', async () => {
+        const { httpSpy, transaction } = context;
+        expect(document.hidden).toBe(false);
+
+        let count = 0;
+        const pollPromise = transaction.poll({
+          delay: 2000,
+          transactionCallBack: () => {
+            count += 1;
+          }
+        });
+
+        for (let i=0; i<4; i++) {
+          await advanceTestTimers();
+        }
+
+        const result = await pollPromise;
+
+        expect(count).toEqual(3);
+        expect(httpSpy).toHaveBeenCalledTimes(4);
+        expect(result.status).toEqual('SUCCESS');
+      });
+
+      it('should not proceed with flow if document is hidden', async () => {
+        const { httpSpy, transaction } = context;
+        expect(document.hidden).toBe(false);
+
+        togglePageVisibility();
+
+        let count = 0;
+        const pollPromise = transaction.poll({
+          delay: 2000,
+          transactionCallBack: () => {
+            count += 1;
+          }
+        });
+
+        // advance the timers so the flow would have succeed in normal circumstances
+        for (let i=0; i<4; i++) {
+          await advanceTestTimers();
+        }
+
+        // ensure flow did not advance, awaits document focus to return 
+        expect(count).toEqual(0);
+
+        togglePageVisibility();
+        for (let i=0; i<4; i++) {
+          await advanceTestTimers();
+        }
+
+        const result = await pollPromise;
+
+        expect(count).toEqual(3);
+        expect(httpSpy).toHaveBeenCalledTimes(4);
+        expect(result.status).toEqual('SUCCESS');
+      });
+
+      it('should pause flow is document is hidden amist polling', async () => {
+        const { httpSpy, transaction } = context;
+        expect(document.hidden).toBe(false);
+
+        let count = 0;
+        const pollPromise = transaction.poll({
+          delay: 2000,
+          transactionCallBack: () => {
+            count += 1;
+          }
+        });
+
+        // advance the timers so the flow would have succeed in normal circumstances
+        for (let i=0; i<4; i++) {
+          await advanceTestTimers();
+          if (i == 1) {
+            // hide document in middle of flow
+            togglePageVisibility();
+          }
+        }
+
+        // ensure flow pauses, awaits document focus to return 
+        expect(count).toEqual(2);
+
+        togglePageVisibility();
+        await Promise.resolve();
+        for (let i=0; i<2; i++) {
+          await advanceTestTimers();
+        }
+
+        const result = await pollPromise;
+
+        expect(count).toEqual(3);
+        expect(httpSpy).toHaveBeenCalledTimes(4);
+        expect(result.status).toEqual('SUCCESS');
+      });
+    });
```

Comment:
Is it worth adding another test case to handle the scenario where document state alternates between hidden and visible multiple times and ensure that polling resumes correctly?

Reply:
added